src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -32,62 +32,33 @@ function LinkIcon({ url }: { url: string }) {
   return <EuiImage size="s" alt="" aria-hidden={true} url={url} />;
 }
 
-export interface NavLink {
-  key: string;
-  label: string;
-  href: string;
-  isActive: boolean;
-  onClick(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
-  category?: AppCategory;
-  isDisabled?: boolean;
-  iconType?: string;
-  icon?: JSX.Element;
-  order?: number;
-  'data-test-subj': string;
+interface Props {
+  link: ChromeNavLink;
+  legacyMode: boolean;
+  appId: string | undefined;
+  basePath?: HttpStart['basePath'];
+  dataTestSubj: string;
+  onClick?: Function;
+  navigateToApp: CoreStart['application']['navigateToApp'];
 }
 
-/**
- * Create a link that's actually ready to be passed into EUI
- *
- * @param navLink
- * @param legacyMode
- * @param currentAppId
- * @param basePath
- * @param navigateToApp
- */
-export function createNavLink(
-  navLink: ChromeNavLink,
-  legacyMode: boolean,
-  currentAppId: string | undefined,
-  basePath: HttpStart['basePath'],
-  navigateToApp: CoreStart['application']['navigateToApp']
-): NavLink {
-  const {
-    legacy,
-    url,
-    active,
-    baseUrl,
-    id,
-    title,
-    disabled,
-    euiIconType,
-    icon,
-    category,
-    order,
-    tooltip,
-  } = navLink;
-  let href = navLink.url ?? navLink.baseUrl;
-
-  if (legacy) {
-    href = url && !active ? url : baseUrl;
-  }
+export function createEuiListItem({
+  link,
+  legacyMode,
+  appId,
+  basePath,
+  onClick = () => {},
+  navigateToApp,
+  dataTestSubj,
+}: Props) {
```

Comment:
NIT: as it's returning a `EUISomething` (`EUIListItem` I guess?) compatible type, I would add an explicit return type to more easily locate a potential type mismatch if the EUI type was to change during an update (the missing return type is also the 'reason' why the `onClick` signature had to be added)